gnome-initial-setup/pages/summary/gis-summary-page.c

```diff
@@ -349,6 +463,22 @@ gis_summary_page_constructed (GObject *object)
   gtk_widget_show (GTK_WIDGET (page));
 }
 
+static void
+gis_summary_page_finalize (GObject *object)
+{
+  GisSummaryPage *page = GIS_SUMMARY_PAGE (object);
+  GisSummaryPagePrivate *priv = gis_summary_page_get_instance_private (page);
+
+  if (priv->clippy_proxy)
+    {
+      g_signal_handler_disconnect (priv->clippy_proxy,
+                                   priv->clippy_proxy_signal_id);
```

Comment:
Since we only have one reference on the object, do you need to disconnect directly? I guess it's okay to be defensive in any case...

Reply:
What if the proxy has internal references due to calls which are in flight? I think it's right to do this.

Reply:
However, it's possible for `clippy_proxy` to be non-`NULL` but `priv->clippy_proxy_signal_id` to be `0` in which case this call will critical.